token-erc20/chaincode-javascript/lib/tokenERC20.js

```diff
@@ -0,0 +1,367 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+*/
+
+'use strict';
+
+const { Contract } = require('fabric-contract-api');
+
+// Define objectType names for prefix
+const balancePrefix = 'balance';
+const allowancePrefix = 'allowance';
+
+// Define key names for options
+const nameKey = 'name';
+const symbolKey = 'symbol';
+const decimalsKey = 'decimals';
+const totalSupplyKey = 'totalSupply';
+
+class TokenERC20Contract extends Contract {
+
+    /**
+     * Return the name of the token - e.g. "MyToken".
+     * The original function name is `name` in ERC20 specification.
+     * However, 'name' conflicts with a parameter `name` in `Contract` class.
+     * As a work around, we use `tokenName` as an alternative function name.
+     *
+     * @param {Context} ctx the transaction context
+     * @returns {String} Returns the name of the token
+    */
+    async tokenName(ctx) {
+        const nameBytes = await ctx.stub.getState(nameKey);
+        return nameBytes.toString();
+    }
+
+    /**
+     * Return the symbol of the token. E.g. “HIX”.
+     *
+     * @param {Context} ctx the transaction context
+     * @returns {String} Returns the symbol of the token
+    */
+    async symbol(ctx) {
+        const symbolBytes = await ctx.stub.getState(symbolKey);
+        return symbolBytes.toString();
+    }
+
+    /**
+     * Return the number of decimals the token uses
+     * e.g. 8, means to divide the token amount by 100000000 to get its user representation.
+     *
+     * @param {Context} ctx the transaction context
+     * @returns {Number} Returns the number of decimals
+    */
+    async decimals(ctx) {
```

Comment:
Since Solidity supports only integer, decimals is only used for display purposes and all calculation in contract is performed on integers in ERC-20. The user interfaces adjust the displayed values according to decimals. How about using decimals() with the same approach in Fabric?

Reply:
sure.